src/prefect/cli/deploy.py

```diff
@@ -802,16 +802,21 @@ async def _run_single_deploy(
                         " YAML file."
                     ),
                 )
+    valid_worker = False
+    if work_pool_name:
+        workers = await client.read_workers_for_work_pool(work_pool_name)
```

Comment:
There's a potential for *a lot* of old workers to show up in this call. Can you update this to include a `status` filter so that only `ONLINE` workers are returned? The API supports that filter but has yet to be added to the client.